com.wso2telco.dep.gw.ussdapi/ussdapigw-synapse/src/main/synapse-config/sequences/modifyUSSDResponseSeq.xml

```diff
@@ -1,16 +1,21 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <sequence name="modifyUSSDResponseSeq" xmlns="http://ws.apache.org/ns/synapse">
-    <class name="org.wso2telco.dep.nashornmediator.NashornMediator">
-        <property name="script" value='
-                var payload= mc.getPayloadJSON();
-                var hubGateway = mc.getProperty("hubGateway");
-                var subResourcePath = mc.getProperty("REST_SUB_REQUEST_PATH");
-                var apiContext = mc.getProperty("REST_API_CONTEXT");
-                var apiVersion = mc.getProperty("VERSION");
-                var subscriptionID = mc.getProperty("subscriptionID");
-                payload.subscription.resourceURL = hubGateway + apiContext + "/" + apiVersion + subResourcePath + "/" + subscriptionID;
-                payload.subscription.callbackReference.notifyURL = mc.getProperty("originalNotifyURL");
-                mc.setPayloadJSON(payload);
-                '/>
+
+    <property name="propertyReplacement"
+              expression="fn:concat(syn:get-property('hubGateway'), syn:get-property('REST_API_CONTEXT'), '/', syn:get-property('VERSION'), syn:get-property('REST_SUB_REQUEST_PATH'), '/', syn:get-property('subscriptionID'))"
+              scope="default" xmlns:fn="http://www.w3.org/2005/xpath-functions"/>
+
+    <class name="com.wso2telco.dep.common.mediation.UtilMediator">
+        <property name="propertyValue" value="singlePropertyReplacement"/>
+        <property name="propertyName" value="resourceURL"/>
+        <property name="propertyPath" value="payload.subscription.resourceURL"/>
+        <property name="msgContextProperty" value="propertyReplacement"/>
+    </class>
+
+    <class name="com.wso2telco.dep.common.mediation.UtilMediator">
```

Comment:
Are these both "modifySendUSSDResponsePayloadSeq.xml" and "modifyUSSDResponseSeq.xml"  sequences for the same reason? If not, it is better to use meaningful sequence names (Eg: modifySendUssdResponse, modifyUssdSubscriptionResponse) because it is difficult to understand the usage of the sequence.

Reply:
resolved